icu4c/source/i18n/messageformat2_parser.cpp

```diff
@@ -1718,15 +1673,18 @@ void Parser::parseUnsupportedStatement(UErrorCode& status) {
     dataModel.addUnsupportedStatement(builder.build(status), status);
 }
 
-// Terrible hack to get around the ambiguity between `matcher` and `reserved-statement`
-bool Parser::nextIsMatch() const {
+// Terrible hack to get around the ambiguity between unsupported keywords
+// and supported keywords
+// (It's convenient that all supported keywords have the same length)
+bool Parser::nextIs(const UChar32 (&keyword)[7]) const {
     for(int32_t i = 0; i < 6; i++) {
```

Comment:
move 6 / 7 into a named constant

Reply:
Done in [3e5b079](https://github.com/unicode-org/icu/pull/3063/commits/3e5b0799dd8ae1ca44b7d5f53e2cdba11238853a)